provider/agent.go

```diff
@@ -284,6 +288,12 @@ func agentResource() *schema.Resource {
 										Description: "The memory usage threshold in percentage at which to trigger an alert. Value should be between 0 and 100.",
 										ForceNew:    true,
 										Required:    true,
+										ValidateDiagFunc: func(i interface{}, s cty.Path) diag.Diagnostics {
+											if i.(int) < 0 || i.(int) > 100 {
+												return diag.Errorf("volume threshold must be between 0 and 100")
+											}
+											return nil
+										},
```

Comment:
`github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation` has a bunch of built-in validations.

```suggestion
										ValidateDiagFunc: validation.IntBetween(0, 100),
```